From: =?utf-8?q?Ernst_Sj=C3=B6strand?= <ernstp@gmail.com>
Date: Thu, 21 Oct 2021 06:39:05 +0200
Subject: Avoid pango assertion on empty title

Pango-CRITICAL **: pango_layout_set_text:
assertion 'length == 0 || text != NULL' failed

Origin: upstream, 0.1.1, commit:754432e3b38356583a9a1270e2b0be7f097968e0
---
 src/plugins/cairo/libdecor-cairo.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/plugins/cairo/libdecor-cairo.c b/src/plugins/cairo/libdecor-cairo.c
index cfc3c83..64138e0 100644
--- a/src/plugins/cairo/libdecor-cairo.c
+++ b/src/plugins/cairo/libdecor-cairo.c
@@ -1134,10 +1134,16 @@ draw_title_text(struct libdecor_frame_cairo *frame_cairo,
 	double text_x, text_y;
 	double text_width, text_height;
 
+	const char *title;
+
+	title = libdecor_frame_get_title((struct libdecor_frame*) frame_cairo);
+	if (!title)
+		return;
+
 	layout = pango_cairo_create_layout(cr);
 
 	pango_layout_set_text(layout,
-			      libdecor_frame_get_title((struct libdecor_frame*) frame_cairo),
+			      title,
 			      -1);
 	pango_layout_set_font_description(layout, frame_cairo->plugin_cairo->font);
 	pango_layout_get_size(layout, &text_extents_width, &text_extents_height);
